service/src/main/kotlin/fi/espoo/evaka/placement/PlacementService.kt

```diff
@@ -314,6 +320,36 @@ private fun Database.Transaction.clearOldPlacements(
         }
 }
 
+private fun Database.Transaction.clearFutureNonPreschoolRelatedPlacements(
+    childId: ChildId,
+    startingFrom: LocalDate,
+) {
+    val futurePlacements = getPlacementsForChildDuring(childId, startingFrom, null)
+    val futureSchoolYearsWithPreschoolPlacements =
+        futurePlacements
+            .filter {
+                listOf(
+                        PlacementType.PREPARATORY,
+                        PlacementType.PREPARATORY_DAYCARE,
+                        PlacementType.PRESCHOOL,
+                        PlacementType.PRESCHOOL_DAYCARE
+                    )
+                    .contains(it.type)
+            }
+            .map {
+                FiniteDateRange(
+                    it.startDate.withMonth(8).withDayOfMonth(1),
+                    it.endDate.withMonth(7).withDayOfMonth(31)
```

Comment:
Doesn't this work in an unexpected way if `startDate` is before 1st of August or `endDate` is after 31st of July?

Reply:
👍 I reworked it a bit from another angle, which I think should take care of the issue.